src/runtime/transaction_execution.zig

```diff
@@ -509,6 +519,113 @@ pub fn executeTransaction(
     };
 }
 
+// TODO: RuntimeTransaction already contains this information which we should use in the future
+// instead of allocating a new array here.
+fn getInstructionDatasSliceForPrecompiles(
```

Comment:
Why did you take this approach in the first place if the returned data is already present in the context that calls this function?